vita3k/io/src/io.cpp

```diff
@@ -619,6 +619,47 @@ int remove_file(IOState &io, const char *file, const std::wstring &pref_path, co
     return 0;
 }
 
+int rename(IOState &io, const char *old_name, const char *new_name, const std::wstring &pref_path, const char *export_name) {
+    auto device = device::get_device(old_name);
+    if (device == VitaIoDevice::_INVALID) {
+        LOG_ERROR("Cannot find device for path: {}", old_name);
+        return IO_ERROR(SCE_ERROR_ERRNO_ENOENT);
+    }
+
+    const auto translated_old_path = translate_path(old_name, device, io.device_paths);
+    if (translated_old_path.empty()) {
+        LOG_ERROR("Cannot translate path: {}", translated_old_path);
+        return IO_ERROR(SCE_ERROR_ERRNO_ENOENT);
+    }
+
+    const auto translated_new_path = translate_path(new_name, device, io.device_paths);
+    if (translated_new_path.empty()) {
+        LOG_ERROR("Cannot translate path: {}", translated_new_path);
+        return IO_ERROR(SCE_ERROR_ERRNO_ENOENT);
+    }
+
+    const auto emulated_old_path = device::construct_emulated_path(device, translated_old_path, pref_path, io.redirect_stdio);
+    if (!fs::exists(emulated_old_path)) {
+        LOG_ERROR("File does not exist at path: {} (target path: {})", emulated_old_path.string(), old_name);
+        return IO_ERROR(SCE_ERROR_ERRNO_ENOENT);
+    }
+
+    const auto emulated_new_path = device::construct_emulated_path(device, translated_new_path, pref_path, io.redirect_stdio);
+
+    LOG_TRACE_IF(log_file_op, "{}: Renaming file {} to {} ({} to {})", export_name, old_name, new_name, emulated_old_path.string(), emulated_new_path.string());
+
+    boost::system::error_code error_code{};
+    fs::detail::rename(emulated_old_path, emulated_new_path, &error_code);
```

Comment:
```suggestion
    fs::rename(emulated_old_path, emulated_new_path, &error_code);
```

fs::detail is not portable (boost only)

Reply:
Other part of the code using it will have to be changed.

Reply:
ah i see hmm but i no can check error code so ? 🤔

Reply:
![image](https://github.com/Vita3K/Vita3K/assets/5671744/b1a6ff67-3339-45e8-a0a1-11393602d7fe)
You should be able to

Reply:
ah yes.. i hgave no understund, it said me no can, i remove, and after i re set and said is ok, visual studio drink

Reply:
ah no çok, it no want i using &error_code but pass with just error_code

Reply:
ok, so done